### PR TITLE
Resolve #10: scm-source.json should be generated at build time

### LINKED ISSUE #10: scm-source.json should be generated at build time

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**

### CODE REVIEW

**Comment by AlexanderYastrebov** [comment]
Should be on top of #7 to pass CI build

File: catwatch-backend/etc/scm/scm-source.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+REV=$(git rev-parse HEAD)
+URL=$(git config --get remote.origin.url)
```
Comment (rwitzel) [inline catwatch-backend/etc/scm/scm-source.sh:3]: This works for both the repo cloned directly and cloned from a GitHub fork, right?

File: catwatch-backend/etc/scm/scm-source.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+STATUS=$(git status --porcelain)
+if [ -n "$STATUS" ]; then
+    REV="$REV (locally modified)"
+fi
+# finally write hand-crafted JSON to scm-source.json
+echo '{"url": "git:'$URL'", "revision": "'$REV'", "author": "'$USER'", "status": "'$STATUS'"}' > target/generated-sources/scm-source.json
```
Comment (rwitzel) [inline catwatch-backend/etc/scm/scm-source.sh:9]: Did you find a specification for the content of such a scm-source.json?

Comment (hjacobs) [inline catwatch-backend/etc/scm/scm-source.sh:9]: you can replace the whole script with our provided Python CLI tool: http://docs.stups.io/en/latest/user-guide/application-development.html#docker

**Comment by AlexanderYastrebov** [comment]
I dont like the idea of installing additional script to build zalando specific scm-source file

**Comment by fogfish** [comment]
I am usually accomplish a project with Makefile file. This make file is used as a wrapper for various things including a 'unnecessary' scripts :-) You can have Make file too. It wraps `scm-source.json` construction while you are building a docker container.

**Comment by AlexanderYastrebov** [comment]
@fogfish Makefile is even worse than custom python script

**Comment by AlexanderYastrebov** [comment]
I'm merging this. Improvements or thoughts on type of scm-source script could be raised as separate issues

**Comment by rwitzel** [comment]
+1

**Pull request merged**